docs/core-migration-guide.md

```diff
@@ -36,6 +36,8 @@ license: |
 
 - Since Spark 4.0, Spark uses `ReadWriteOncePod` instead of `ReadWriteOnce` access mode in persistence volume claims. To restore the legacy behavior, you can set `spark.kubernetes.legacy.useReadWriteOnceAccessMode` to `true`.
 
+- Since Spark 4.0, Spark uses `~/.ivy2.5.2` as Ivy user directory by default to isolate the existing systems from Apache Ivy's incompatibility. To restore the legacy behavior, you can set `spark.jars.ivy` to `~/.ivy2`.
```

Comment:
Will it need to be changed again if we upgrade to use ivy 2.5.3 or 2.6.x in the future? Or can the name of this directory be:

```
.ivy2.5.2_and_above
.ivy2.5.2_plus
.ivy2.5.2_upwards
.ivy2.5.2_or_higher
```
?

Reply:
I also thought like that. Something like `.ivy2.5.2+`.

After receiving your comment, I'm rethinking about that. 

The bottom line is that the compatibility and release cycle depends on the Apache Ivy community. 

- `.ivy2.5.2` literally means Apache Ivy format written by Apache Ivy 2.5.2 .
  - If Apache Ivy 2.5.3 is not going to introduce any new change, it's still Apache Ivy 2.5.2-format.
  - If Apache Ivy 2.5.3 breaks the format again, we need to use `.ivy2.5.3` at that time.

- In addition, if we use `ivy2.5.2_or_higher`, it could be an over-claim because Apache Spark community is unable to guarantee any compatibility for Apache Ivy 2.5.3 or higher which implies the naming.

Reply:
Like `.ivy2`, we don't need to change this until the next Apache Ivy breaking change, @LuciferYang .

Reply:
SGTM